### DISCUSSION TITLE
Player side selection for Versus Co-op Mode

### DISCUSSION THREAD

**Discussion by pokeshark** [ikemen-engine/Ikemen-GO#858] [Feature Request]
Is it possible to add a player side selection before selecting characters in Versus Co-op Mode?
Would be very useful if we want to rotate players and teams.

Example from SFxT: https://i.imgur.com/jdRg3aj.png
Example from MVCI: https://i.redd.it/niclz8suevuz.jpg

Thank you for adding Tag mode with MVCI ruleset. It's really fun!

**Comment by pokeshark** [comment]
until this gets hopefully implemented, i'm having trouble testing the Tag mode with 2 controllers and a keyboard.

P1 and P2 on keyboard overlap with P1 and P2 from controllers, controlling the same characters.
the controls for P3 and P4 on keyboard don't seem to be working in Tag mode.

pairing the controllers on the same team (vs a team of keyboard players) also doesn't seem possible.

**Comment by K4thos** [comment]
that's a limitation of current implementation. I do agree current system is bad. I'm planning to rewrite how it all works in 0.99 using Tekken 7 as a reference. Auto detection of gamepad mappings based on SDL database, configuration on per controller basis rather than per player, no keyboard and gamepad sharing by the same player, controller and side selection at the start of select screen.

**Comment by pokeshark** [comment]
oh, I thought I was doing something incorrectly.
good to hear there are plans down the line to address this. thanks for reply.
